estimagic/visualization/comparison_plot.py

```diff
@@ -0,0 +1,440 @@
+from bokeh.models import HoverTool
+from bokeh.models import TapTool
+from bokeh.models import Title
+from bokeh.models.callbacks import CustomJS
+from bokeh.models.widgets import CheckboxGroup
+from bokeh.plotting import figure
+from bokeh.plotting import show
+
+from estimagic.optimization.utilities import index_element_to_string
+
+
+def comparison_plot(
+    res_dict, color_dict=None, height=None, width=600, point_estimate_plot_kwargs=None
+):
+    """Make a comparison plot from a dictionary containing optimization results.
+
+    Args:
+        res_dict (dict): The keys are the names of different models.
+            Each value is a dictinoary with the following keys and values:
+                - result_df (pd.DataFrame):
+                    params DataFrame returned by maximize or minimize.
+                - model_class (str, optional):
+                    name of the model class to which the model belongs.
+                    This determines the color and checkbox entries with
+                    which model classes can be selected and unselected.
+
+        color_dict (dict):
+            mapping from the model class names to colors.
+
+        height (int):
+            height of the plot.
+
+        width (int):
+            width of the plot.
+    """
+    df = _df_with_all_results(res_dict)
+    heights = _determine_plot_heights(df, height)
+    lower, upper, rect_widths = _create_bounds_and_rect_widths(df)
+    _add_plot_specs_to_df(df, rect_widths, lower, upper, color_dict)
+
+    source_dict, figure_dict, glyph_dict = _create_comparison_plot_components(
+        df=df,
+        heights=heights,
+        lower=lower,
+        upper=upper,
+        rect_widths=rect_widths,
+        width=width,
+    )
+
+    plots_with_callbacks = _add_callbacks(
+        source_dict=source_dict,
+        figure_dict=figure_dict,
+        glyph_dict=glyph_dict,
+        model_classes=sorted(df["model_class"].unique()),
+    )
+
+    grid = gridplot(plots_with_callbacks, toolbar_location="right", ncols=1)
+    show(grid)
+    return df, grid, plots_with_callbacks
+
+
+# ===========================================================================
+# DATA PREP FUNCTIONS
+# ===========================================================================
+
+
+def _df_with_all_results(res_dict):
+    """
+    Build the DataFrame combining all DataFrames from the results dictionary.
+
+    Args:
+        res_dict (dict): see comparsion_plot docstring
+    """
+    df = pd.DataFrame()
+    for model, model_dict in res_dict.items():
+        result_df = _prep_result_df(model_dict, model)
+        df = df.append(result_df, sort=False)
+
+    if "group" not in df.columns:
+        df["group"] = "all"
+    if "conf_int_upper" not in df.columns:
+        df["conf_int_upper"] = np.nan
+    if "conf_int_lower" not in df.columns:
+        df["conf_int_lower"] = np.nan
+
+    return df.reset_index(drop=True)
+
+
+def _prep_result_df(model_dict, model):
```

Comment:
In case a dataframe has no column ``final_value`` can we look for ``value`` instead? My use case is that if I have a model description which is not directly out of estimagics min max functions but an input, it does not have ``final_value`` but ``value``. Still, I might one to compare my parametrizations.

Reply:
You are right, it should expect ``value`` instead of ``final_value``. Fixed in 504d93d28d61c9b123041da886811b7151cb9c58.